dicompylercore/dicomparser.py

```diff
@@ -625,6 +638,16 @@ def GetStructureCoordinates(self, roi_number):
                             plane['data'] = \
                                 self.GetContourPoints(c.ContourData)
 
+                            # Verify if all z-coordinates in this plane
+                            # are identical
+                            if not self.verify_z_coordinates(plane['data']):
```

Comment:
Suggestion: could avoid `list` changes/creations by working with the unflattened `ContourData` directly  and using the `islice` iterator to skip every third (untested):
```python
from itertools import islice

z0 = c.ContourData[2]
if not all(z == z0 for z in islice(c.ContourData, 5, None, 3)):
    ...
```
And this is simple enough to not need a separate function, I think.

Reply:
Yes, thank you that is much cleaner. I tested it and it works.